specification.md

```diff
@@ -5157,43 +5225,37 @@ Workflow constants should not have access to [Workflow secrets definitions](#Wor
 
 ### Workflow Secrets
 
-Secrets allow you access sensitive information, such as passwords, OAuth tokens, ssh keys, etc
-inside your [Workflow Expressions](#Workflow-Expressions). 
+Secrets allow you access sensitive information, such as passwords, OAuth tokens, ssh keys, etc.
+They can be used in [Workflow Expressions](#Workflow-Expressions).
 
 You can define the names of secrets via the [Workflow top-level "secrets" property](#Workflow-Definition-Structure), 
 for example:
 
 ```json
-"secrets": ["MY_PASSWORD", "MY_STORAGE_KEY", "MY_ACCOUNT"]
+"secrets": ["MY_ACCOUNT", "MY_PASSWORD", "urn:example:tenant:accessToken"]
 ```
 
-If secrets are defined in a Workflow definition, runtimes must assure to provide their values
-during Workflow execution.
+If secrets are only named in a Workflow definition, runtimes must assure to 
+provide their values during execution.
 
-Secrets can be used only in [Workflow expressions](#Workflow-Expressions) under the `SECRETS` namespace. 
-This is reserved namespace that should only be allowed for values defined by the `secrets` property.
-
-Here is an example on how to use secrets and pass them as arguments to a function invocation:
+Secrets can be used only in [workflow expressions](#Workflow-Expressions).
+The default expression language JQ provides secrets through the reserved 
+`$SECRETS` variable. Here is an example on how to use secrets and pass them as 
+arguments to a function invocation:
 
 ```json
-"secrets": ["AZURE_STORAGE_ACCOUNT", "AZURE_STORAGE_KEY"],
-
-...
-        
 {
   "refName": "uploadToAzure",
     "arguments": {
-      "account": "${ $SECRETS.AZURE_STORAGE_ACCOUNT }",
-      "account-key": "${ $SECRETS.AZURE_STORAGE_KEY }",
-      ...
+      "user": "${ $SECRETS.MY_ACCOUNT }",
+      "pass": "${ $SECRETS.MY_PASSWORD }",
+      "token": "${ $SECRETS.\"urn:example:tenant:accessToken\" }"
```

Comment:
this is so ugly to me :( i dont think there is a reason for it really @cdavernas am i missing something?

Reply:
@tsurdilo What is ugly? What is done with the `token` property? If yes, I agree. From where I stand I do not see any need for supporting such ugly (and dangerous) practices.

Reply:
@cdavernas yes, the "token" property value

Reply:
That's basic JQ. There's currently no restriction on the string - neither in the secrets definition nor in JQ. Not sure what it is you don't want to support? Do you want to restrict secret names to be capital letter snake case?